tests/std/tests/P0408R7_efficient_access_to_stringbuf_buffer/test.cpp

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <memory_resource>
+#include <sstream>
+#include <string>
+#include <string_view>
+#include <utility>
+
+#include <test_death.hpp>
+
+using namespace std;
+
+constexpr const char empty_string[] = "";
+constexpr const char small_string[] = "a";
+constexpr const char large_string[] =
+    "This is a long long long long long long long long string to avoid small string optimization.";
+
+template <typename Stream>
+struct test_rvalue {
+    void operator()(const string& init_value) {
+        string buffer{init_value};
+        size_t res = buffer.capacity();
+        Stream stream{move(buffer)};
+        assert(stream.view() == init_value);
+        assert(stream.str() == init_value);
+        assert(stream.view() == init_value);
+        assert(stream.rdbuf()->get_allocator() == init_value.get_allocator());
+        // Move out the buffer, the underlying buffer should be empty.
+        buffer = move(stream).str();
+        if (buffer == string_view{large_string}) {

```

Comment:
The `string buffer` can be directly compared to `const char [N]`, and constructing a temporary `string_view` provides no advantages:
```suggestion
        if (buffer == large_string) {
```